keylime/ima/ast.py

```diff
@@ -29,11 +29,15 @@
 
 
 def get_START_HASH(hash_alg: Hash) -> bytes:
-    return codecs.decode(b"0" * (hash_alg.get_size() // 4), "hex")
+    hsize = hash_alg.get_size()
```

Comment:
You are doing this to make pyright happy, right? It complains that `hsize` could be `None`? A hash whose size could be `None`...  hm... ? I am surprised it doesn't complain that one couldn't called `get_size()` on  a `NoneType`.

Reply:
I am not following you on the "missing" complaint: what do you mean calling `get_size()` on  a `NoneType`? Anyway yes, I am "fixing" the assumption that `hash_alg.get_size()` is not a `NoneType`, as that would break the `//` application type-wise.

Reply:
I think that was the second time that we discussed this same point. Seems that the reason is this:

https://github.com/keylime/keylime/blob/5a7c5e3d3dde1d35fec688323744fe9530e585b8/keylime/common/algorithms.py#L48

_HASH_SIZE is a dict and the size is checked as a query on it. I would be in favor of not making it optional, and change the class in a way that if the hash family is not recognized, fail.

But for this PR, that is about adding type hints without changing semantic, I think that what @mdrocco did it is OK

Reply:
thanks @aplanas for clarifying, in fact I saw the same pattern here and there while applying type fixes/hints, we might want to take an extra step and remove some of those `Optional` as further refinement, but for the sake of minimalism I would not include that in this PR.

Reply:
> But for this PR, that is about adding type hints without changing semantic, I think that what @mdrocco did it is OK

As long as we remember that the added asserts are hiding the issue...

Reply:
An issue can be created, indeed

Reply:
PR #1189 should resolve this and the issue below.

Reply:
I dropped the assert